source/common/ssl/context_impl.cc

```diff
@@ -222,7 +222,15 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const ContextConfig& config, TimeS
     bio.reset(BIO_new_mem_buf(const_cast<char*>(tls_certificate.privateKey().data()),
                               tls_certificate.privateKey().size()));
     RELEASE_ASSERT(bio != nullptr, "");
-    bssl::UniquePtr<EVP_PKEY> pkey(PEM_read_bio_PrivateKey(bio.get(), nullptr, nullptr, nullptr));
+
+    bssl::UniquePtr<EVP_PKEY> pkey = nullptr;
+    if (!tls_certificate.password().empty()) {
+      pkey = bssl::UniquePtr<EVP_PKEY>(PEM_read_bio_PrivateKey(
+          bio.get(), nullptr, nullptr, const_cast<char*>(tls_certificate.password().c_str())));
+    } else {
+      pkey =
+          bssl::UniquePtr<EVP_PKEY>(PEM_read_bio_PrivateKey(bio.get(), nullptr, nullptr, nullptr));
+    }
```

Comment:
This might be slightly more readable (but I don't feel too strongly about it, so feel free to leave it as-is):
```
bssl::UniquePtr<EVP_PKEY> pkey(PEM_read_bio_PrivateKey(
    bio.get(), nullptr, nullptr,
    !tls_certificate.password().empty() ? const_cast<char*>(tls_certificate.password().c_str())
                                        : nullptr
    ));
```

Reply:
Addressed in 0b2d916.